### ISSUE TITLE
Time issues

### ISSUE THREAD

**Issue by jhardenberg** [DestinE-Climate-DT/AQUA#74]
This is connected to the 'data model' discussion. 

TL;DR: time coordinate for accumulated and mean-rate convention for ECMWF/grib is different from CMOR and can lead to issues. We discuss options.

The ECMWF convention for accumulated variables or mean rate variables is that 'valid_time' (which commonly is adopted for 'time' dimension) refers to the END of the accumulation period. See for example [this discussion](https://confluence.ecmwf.int/pages/viewpage.action?pageId=85402030) for ERA5.
So for example precipitation `tp` at 12:00 hours, saved with a frequency of 3h, refers to accumulated precipitation between 9 and 12. Actually it may also refer to precipitation accumulated since the initial forecast_init_time (nextGEMS data on levante are saved this way) - but let's assume that these data have been 'decumulated' to be accumulated just over output time steps.  The same is true for mean rate variables, 'tprate' would follow the same convention and `tprate' at 12:00 in m/s is average precip between 9 and 12.
The big advantage of this convention is that all data in a dataset which we open with our reader (even if it mixes instantaneous and accumulated variables) have exactly the same time. So their time dimension is common and aligned. 
On the other hand (personal opinion) this is quite terrible when working particularly with mean rates (m/s, W/m2 etc): 
- What happens if we average these data? If we select precipitation saved at hourly frequency for a specific day, we will 'forget' the precipitation between 23 and 24 hours, since it is actually marked with time=00:00 of the following day.
- If we average mean rates what should be the date and time of the result? according to the ECMWF convention it should actually again be the end of the averaging period. Imagine a monthly average which has date 00:00 of the first day of the **following month.** ... quite confusing
- Further, if we average variables and we decide to also average times, the resulting time is not much meaningful. The daily average of precipitation at hourly frequency following this convention will have time=11:30 = (0+1+..23)/24. If we include (correctly) the missing last hour and we drop the data at time=0 (which pertains to the previous day) we end up with an 'average time' of 12:30 PM ... again not good.
- The CMOR convention which uses time centered at the middle of the averaging period does not present any of these issues.

So **for the purposes of AQUA** it could be useful to have a time axis which follows, for cumulated and mean-rate variables, the 'center of period' convention. On the other hand we cannot simply shift the 'time' coordinate of mean-rate and accumulated variables because this would lead to a dataset in which we have some variables (e.g. 2t) which have one time axis and others (e.g. tp) which have another, which, as far as I know is not possible (like a netcdf file, coordinates in a Dataset should be shared). 
I actually see two solutions:
1) have the reader actually return dictionaries of dataarrays or datasets instead of datasets (which then could have different coordinates) ... this would lead to quite some complication in working with the data and I would not favour this option.
2) create a new additional coordinate in the dataset which is 'shifted' the right amount to be at the center of accumulation period for accumulated and mean-rate variables, to be later used by users in the analysis if they so wish. So "time" would still refer to the end of the accumulation period, and this would be the coordinate associated with the 'time' dimension, but this additional new coordinate (also depending on the 'time' dimension), which maybe we could call "center_time" or similar (ideas/suggestions?) would point to the center of the cumulation period and could be used for plots etc.
Options for names: center_time, ctime, atime .... I would vote for "ctime". 
Adding also a "time_bounds" coordinate  for completeness would also be neat.

Sorry for the length. Other suggestions, comments/opinions?
